Misc/NEWS.d/next/Core_and_Builtins/2024-12-15-16-56-26.gh-issue-126085.BvyEXk.rst

```diff
@@ -0,0 +1,2 @@
+Add the ``tp_iter`` slot to :class:`typing.TypeAliasType` to allow star
```

Comment:
I think this would be good in a What's New so that users know they can use this form. And I would rather omit `tp_iter` and simply say that `TypeAliasType` support star unpacking (possibly with some examples in the docs?)

Reply:
Adding this to What's New sounds a like a good idea! I'll update the PR later today :)

> possibly with some examples in the docs?

I could add another example here: https://docs.python.org/3.14/library/typing.html#typing.TypeAliasType
using `__iter__` as a new method subsection perhaps?